manager/volume.go

```diff
@@ -622,6 +625,38 @@ func (m *VolumeManager) Expand(volumeName string, size int64) (v *longhorn.Volum
 		return v, nil
 	}
 
+	pvcName := v.Status.KubernetesStatus.PVCName
+	resourceSize := resource.MustParse(strconv.FormatInt(size, 10))
+	if pvcName != "" {
+		pvcNamespace := v.Status.KubernetesStatus.Namespace
+		pvc, err := m.ds.GetPersisentVolumeClaim(pvcNamespace, pvcName)
+		if err != nil {
+			return nil, err
+		}
+
+		// Should change to pvc.Status.Capacity.Storage(), once upgrade API to v0.18.x
+		if pvc.Status.Capacity.StorageEphemeral().Cmp(resourceSize) > 0 {
+			return nil, fmt.Errorf("cannot expand volume %v with current size %v to a smaller size %v", v.Name, v.Spec.Size, size)
+		}
+
+		// Should change to pvc.Status.Capacity.Storage(), once upgrade API to v0.18.x
+		if pvc.Status.Capacity.StorageEphemeral().Cmp(resourceSize) == 0 {
+			logrus.Infof("Volume %v expansion is not necessary since current size %v == %v", v.Name, v.Spec.Size, size)
+			return v, nil
+		}
+
+		pvc.Spec.Resources = apiv1.ResourceRequirements{
+			Requests: apiv1.ResourceList{
+				apiv1.ResourceStorage: resourceSize,
+			},
+		}
+
+		_, err = m.ds.UpdatePersisentVolumeClaim(pvcNamespace, pvc)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Update the PVC only if `resourceSize` is greater than the current PVC spec size.

Reply:
With previous conditions check, only request pvc size greater than current size will run here.